presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -521,6 +525,9 @@ private static void addLookupOuterDrivers(LocalExecutionPlanContext context)
         private final List<DriverFactory> driverFactories;
         private final Optional<IndexSourceContext> indexSourceContext;
 
+        private TupleDomain<DynamicFilters.Descriptor> dynamicFilter = TupleDomain.all();
```

Comment:
I think dynamic filter should be passed to subcontext, e.g. in collocated joins there could be two scans of same table (self join)

Reply:
`TupleDomain<DynamicFilters.Descriptor> dynamicFilter` is not longer a planning object. I think logic from context should be extracted to some `LocalDynamicFiltersCollector` that would contain both `intersectDynamicFilter` and `getDynamicFilterSupplier` methods.

Reply:
Extracted the functionality into a separate class, allowing subcontexts to use the it as well.